File: WEEK_3_Python/Q4.py
class Node:
    def __init__(self, data):
        self.data, self.next = data, None

def search(head, key):
    cur = head
    while cur:
        if cur.data == key: return True
        cur = cur.next
    return False

# Example
head = Node(1); head.next = Node(2); head.next.next = Node(3)
print("Found" if search(head, 2) else "Not Found")   

File: WEEK_3_C/Q4.c
#include <stdio.h>
#include <stdlib.h>

struct Node{ int data; struct Node* next; };

int main(){
    struct Node *head=NULL,*t;
    int cnt=0;
    for(int i=1;i<=5;i++){ t=(struct Node*)malloc(sizeof(struct Node)); t->data=i; t->next=head; head=t; }
    t=head;
    while(t){ cnt++; t=t->next; }
    printf("Count=%d\n",cnt);
    return 0;
}